### ISSUE TITLE
Memory leak regress in 0.9.3 due to new runFoldScope

### ISSUE THREAD

**Issue by mpilquist** [functional-streams-for-scala/fs2#812]
This program leaks memory:

```scala
object ConstantEvalSanityTest extends App {
  import TestUtil.S
  Stream.constant(()).flatMap { _ => Stream.eval(Task.delay(())) }.run.unsafeRun
}
```

Spotted by @AusCoder.

Caused by https://github.com/functional-streams-for-scala/fs2/pull/808. Straightforward fix is to revert #808 and release 0.9.4 and then fix for real on master / 1.0.

**Comment by pchlupacek** [comment]
+1 pls if possible today/tomorrow :-)

**Comment by pchiusano** [comment]
👍 for just reverting and saving the fix for 1.0

**Comment by mpilquist** [comment]
Fixed -- #808 was fine on series/1.0. When I backported it to series/0.9, I messed up the merge.
